shuup/campaigns/models/campaigns.py

```diff
@@ -224,6 +224,15 @@ def __str__(self):
         return force_text(_("Basket Campaign: %(name)s" % dict(name=self.name)))
 
     def save(self, *args, **kwargs):
+        if self.coupon:
+            code_count_for_shop = BasketCampaign.objects.filter(
```

Comment:
This checkout should also be done on Coupon save right? As we allow merchants to change the coupon code it is possible to have this situation by just renaming the code :)

Also, the `code_count_for_shop.count() == 1` doesn't seem right.. I would suggest to use of `.exist()` if the queryset returns more than 1, I know, that is nearly impossible, but the `exist()` make more sense for me when reading the code.